crates/wast/src/ast/types.rs

```diff
@@ -104,13 +72,68 @@ impl<'a> Parse<'a> for ValType<'a> {
             })
         } else if l.peek::<kw::exnref>() {
             parser.parse::<kw::exnref>()?;
-            Ok(ValType::Exnref)
+            Ok(ValType::Ref(RefType::Exn))
         } else if l.peek::<kw::eqref>() {
             parser.parse::<kw::eqref>()?;
-            Ok(ValType::Eqref)
+            Ok(ValType::Ref(RefType::Eq))
         } else if l.peek::<kw::i31ref>() {
             parser.parse::<kw::i31ref>()?;
-            Ok(ValType::I31ref)
+            Ok(ValType::Ref(RefType::I31))
+        } else {
+            Err(l.error())
+        }
+    }
+}
+
+/// The reference value types for a wasm module.
+#[allow(missing_docs)]
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone)]
+pub enum RefType<'a> {
+    Func,
```

Comment:
Mind adding a few comments to the variants here? The first two I think are reference-types proposal types, the next is exception handling, and everything else is the gc proposal, right?

Reply:
Yes, that's the case. I've added some comments explaining the types.